Adds 12 dof version of solo and display demo script

jviereck opened the pull request:
This adds an extra degree of freedom at the shoulder of the solo robot. This version of the robot called "solo12" (for 12 DOF) at this point.

Besides adding the extra DOF, this also adds 150 g per each new DOF = 600 g of total net weight for the robot.

The order of the joints is: HAA, HFE, KFE.

![solo12](https://user-images.githubusercontent.com/191719/64295634-9b7c1380-cf3f-11e9-8bfb-c7ffd7d409a1.png)

jviereck committed:
Adding solo12: Extra DOF at the shoulders, 150 g extra at each shoulder. Including demo file to display robot.

jviereck commented:
Updated the PR:
* Increased the weight for the solo12 by another 200 g due to @fgrimminger  suggestion (the extra weight was added to the solo12 body)
* Fixed alignment of the foot sensors. See new screenshot. This is also fixed for the 8 DOF solo robot (as the fix is done in the leg.xacro file).

![image](https://user-images.githubusercontent.com/191719/64367785-ddac6000-cfe6-11e9-91f0-630b4d61c6a6.png)

MaximilienNaveau commented on python/robot_properties_solo/config.py:
```diff
@@ -55,25 +91,70 @@ class SoloConfig:
+    )
+
+    # The inertia of a single blmc_motor
+    motor_inertia = 0.0000045
```
Why adding this again in Solo12?
idem for the gear ratio...
Alle these data are the same no?

jviereck replied:
The name is defined but it cannot be used during the initialization of the class.

Here is an example [1]: 

```
class A(object):
  foo = 1
  bar = foo

class B(A):
  baz = foo

B()
> NameError: name 'foo' is not defined
```

Therefore, I've ended up copying some of the definitions.

Do you see a way to have the class variables and avoid the duplications? I was thinking about changing the class variables to instance variables (assign them in an `__init__` call). This would then change the way we use the class right now (e.g. a `SoloConfig.motor_inertia` would become `config = SoloConfig(); config.motor_inertia`.

[1]: https://colab.research.google.com/drive/1uOFFt5lPYsnQRJyTzNryGOlV0dszlaWP

MaximilienNaveau commented on python/robot_properties_solo/config.py:
```diff
@@ -55,25 +91,70 @@ class SoloConfig:
-    # ctrl_manager_current_to_control_gain I am not sure what it does so 1.0.
-    ctrl_manager_current_to_control_gain=1.0
+class Solo12Config(SoloAbstract):
+    # name that is used by every other entities. I do not exactly why...
```
Can we remove this "I do not know exactly why?"

MaximilienNaveau commented on python/robot_properties_solo/config.py:
```diff
@@ -55,25 +91,70 @@ class SoloConfig:
+    nb_joints = robot_model.nv - 6
+
+    # mapping between the ctrl vector in the device and the urdf indexes
+    urdf_to_dgm = (0,1,2,3,4,5,6,7,8,9,10,11)
```
range(12)?

jviereck replied:
Changed this to `tuple(range(12))` and the above for `SoloConfig` to `tuple(range(6))`.

MaximilienNaveau commented on xacro/leg.xacro:
```diff
@@ -11,43 +11,92 @@
   <xacro:property name="HFE_2_KFE_z" value="${160.00 * 0.001}" />
 
   <!-- To check this data to get the center of the foot properly -->
```
If the data has been checked you should remove this comment as well. Thanks for checking this BTW!

jviereck replied:
Done.

MaximilienNaveau reviewed (changes requested):
Can you fix the small things? and then you can self merge.
We should add a documentation about these package. Because the doc is not yet in place we pass on this pull request. But it would be nice to ensure we have some how-to in a doxygen here. typically for the python display and config.

jviereck commented:
Hi Max, thanks for the review. I've addressed most of your comment, please let me know about the open issue with the class variables for subclassed `SoloConfig12`. 

Sorry for doing a force push with the changes (I am used to some other review system I used lately...).

jviereck commented:
@MaximilienNaveau , I merged this PR with the current code as we needed it for some project.

Please let me know if you want me to change it with the open question about using class variables.
